src/nvim/testdir/test_eval_stuff.vim

```diff
@@ -11,3 +11,17 @@ endfunction
 func Test_catch_return_with_error()
   call assert_equal(1, s:foo())
 endfunc
+
+func Test_nocatch_restore_silent_emsg()
+  silent! try
+    throw 1
+  catch
+  endtry
+  echoerr 'wrong'
+  let c1 = nr2char(screenchar(&lines, 1))
+  let c2 = nr2char(screenchar(&lines, 2))
+  let c3 = nr2char(screenchar(&lines, 3))
+  let c4 = nr2char(screenchar(&lines, 4))
+  let c5 = nr2char(screenchar(&lines, 5))
+  call assert_equal('wrong', c1 . c2 . c3 . c4 . c5)
```

Comment:
```
function RunTheTest[32]..Test_nocatch_restore_silent_emsg line 11: Expected 'wrong' but got '     '
```

Reply:
What's the error if you change it to `&lines - 1` or `&lines + 1`?

FWIW the test works if I run this script on HEAD.

Reply:
It fails on gvim using https://github.com/vim/vim-appimage. How did it pass on Vim?